### PR TITLE
FMRF-16: Add FMR app skeleton

### PR DESCRIPTION
**Pull request by dk4g**
### What?
[FMRF-16](https://collaboration.homeoffice.gov.uk/jira/browse/FMRF-16)

Adds the necessary files to create the eVisa - Find my reference form skeleton. This includes the Node.js/HOF and Docker config files needed to build and run a HOF app, plus the basic folder structure for the FRM app itself with minimal content added.

Also includes housekeeping files such as README update, pull request template and ignore files for Docker and Git.

### Why?
This skeleton should allow a developer to run a local version of this app with either Node.js or docker-compose, and work on page specific tickets for the form.

### Anything else?
This PR does not include HOF devops setup such as the /bin, /kube folders or drone.yml. These can be appended to this feature or added as a new feature later. The purpose of this ticket is to enable local development of form pages.

### CODE REVIEW

File: apps/fmr/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const hof = require('hof');
+const Summary = hof.components.summary;
+const clearSession = require('./behaviours/clear-session');
```
Comment (jamiecarterHO) [inline apps/fmr/index.js:3]: You already import the HOF clearSession component in hof.settings.json which can be used with `clearSession: true` on the last step: https://github.com/UKHomeOffice/additional-security-checks/blob/master/apps/asc/index.js#L691

If you prefer this behaviour then no problem

File: apps/fmr/views/layout.html
Comment (jamiecarterHO) [inline apps/fmr/views/layout.html:1]: Is this file necessary? For the start page columns perhaps?

File: apps/fmr/views/partials/items-table.html
Comment (jamiecarterHO) [inline apps/fmr/views/partials/items-table.html:1]: We need this partial?

File: apps/fmr/views/partials/summary-table.html
Comment (jamiecarterHO) [inline apps/fmr/views/partials/summary-table.html:1]: We need this partial for this form?

File: config.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  env: env,
+  govukNotify: {
+    notifyApiKey: process.env.NOTIFY_KEY,
+    caseworkerEmail: process.env.CASEWORKER_EMAIL,
+    userConfirmationTemplateId: process.env.USER_CONFIRMATION_TEMPLATE_ID,
+    businessConfirmationTemplateId: process.env.BUSINESS_CONFIRMATION_TEMPLATE_ID
+  },
+  hosts: {
+    acceptanceTests: process.env.ACCEPTANCE_HOST_NAME || `http://localhost:${process.env.PORT || 8080}`
+  },
```
Comment (jamiecarterHO) [inline config.js:22]: Don't think we need this config

File: config.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      'image/jpg',
+      'image/png',
+      'image/gif'
+    ],
+    documentCategories: {
+      'identity-documents': {
+        limit: 6,
+        limitValidationError: 'maxIdDocsUploads'
+      }
+    }
```
Comment (jamiecarterHO) [inline config.js:41]: is this config relevant to uploads in this form?

File: server.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+settings = Object.assign({}, settings, {
+  behaviours: settings.behaviours.map(require),
+  routes: settings.routes.map(require)
+});
+
+const app = hof(settings);
+
+app.use((req, res, next) => {
+  res.locals.htmlLang = 'en';
```
Comment (jamiecarterHO) [inline server.js:16]: no feedback ;-D

File: test/unit/utils.test.js
Comment (jamiecarterHO) [inline test/unit/utils.test.js:1]: This can be used in the summary or notification emails?

**Review by jamiecarterHO — COMMENTED** [review]
Few questions mainly about whether we can get rid of some bits. 
Lots of Github complaining about no newline at end of file.
Should we update Node 20 LTS version?

File: apps/fmr/views/layout.html
Comment (dk4g) [reply apps/fmr/views/layout.html:1]: yes, for a few reasons:
- page title has a different structure (<H1> - <SERVICE NAME> - GOV.UK)
- added side panel 
- added fullwidth header
- added static section for "Need Help?"

last to three points are covered in #2 and #3

File: apps/fmr/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    },
+    '/nationality': {
+      next: '/date-birth/',
+      fields: []
+    },
+    '/date-birth/': {
+      next: '/upload-photo/',
+      fields: []
+    },
+    '/upload-photo/': {
```
Comment (jamiecarterHO) [inline apps/fmr/index.js:32]: extra slashes at end of routename, but I see you fixed in next PR

File: config.js (hunk 3 above)
Comment (dk4g) [reply config.js:41]: the requirements are not clear if there should be a limit on uploaded files in this category, I'll confirm this as a part of [FMRF-18](https://collaboration.homeoffice.gov.uk/jira/browse/FMRF-18) 
but can remove it for now to avoid confusion.

Comment (dk4g) [reply config.js:41]: Confirmed that the maximum limit is one file, this will remain and be covered in a separate PR.

**Review by jamiecarterHO — APPROVED** [review]
Looks good.

Just leaving a note that we chose to use the AMD64 manifest digest for the base Node image, instead of the multiplatform 'index digest'
